### PR TITLE
chore(chat): get rid of takeUntil and race conditions when accepting share invitation (Issue #2599)

### PR DESCRIPTION
**Pull request by Alexander-Kezik**
**Description:**

get rid of takeUntil and race conditions when accepting share invitation

Issues:

- Issue #2599

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2615.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2615.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/store/conversations/conversations.epics.ts [hunk 1]
```diff
@@ -128,62 +128,59 @@ import uniq from 'lodash-es/uniq';
...
+      }
 
-const getSelectedConversationsEpic: AppEpic = (action$, state$) =>
-  action$.pipe(
-    filter(ConversationsActions.getSelectedConversations.match),
-    switchMap(({ payload }) =>
-      ConversationService.getSelectedConversationsIds().pipe(
+      const previousRoute = UISelectors.selectPreviousRoute(state$.value);
+      const shouldCreateNewConv =
+        previousRoute &&
```
Comment (IlyaBondar) [inline apps/chat/src/store/conversations/conversations.epics.ts]:
```suggestion
        !previousRoute ||
```

**Review by IlyaBondar — CHANGES_REQUESTED** [review]

File: apps/chat/src/store/conversations/conversations.epics.ts [hunk 2]
```diff
@@ -199,37 +196,39 @@ const getSelectedConversationsEpic: AppEpic = (action$, state$) =>
...
             selectedIds: of(selectedIds),
           });
         }),
         map(({ selectedConversations, selectedIds }) => {
-          const conversations = selectedConversations
-            .filter(
-              (conv) =>
-                !!conv &&
-                (!payload?.createNew ||
-                  !conv.messages.filter((m) => m.role !== Role.System).length),
```
Comment (IlyaBondar) [inline apps/chat/src/store/conversations/conversations.epics.ts:212]: without this condition now (w/o #2584) new conversation will be created each time if even selected one already is empty

**Review by denys-kolomiitsev — DISMISSED** [review]

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2615.nightly-test.deltixhub.io
>E2E tests status: success

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**